src/effects/effects_base/internal/effectsactionscontroller.cpp

```diff
@@ -38,6 +43,59 @@ void EffectsActionsController::repeatLastEffect()
     effectExecutionScenario()->repeatLastProcessor();
 }
 
+void EffectsActionsController::addRealtimeEffect(const muse::actions::ActionData& args)
+{
+    IF_ASSERT_FAILED(args.count() == 2) {
+        return;
+    }
+
+    const auto project = globalContext()->currentProject();
+    IF_ASSERT_FAILED(project) {
+        // Command issued without an open project ?..
+        return;
+    }
+
+    const auto effectId = args.arg<au::audio::EffectId>(0);
+    const auto trackId = args.arg<au::audio::TrackId>(1);
+    audioEngine()->addRealtimeEffect(*project, trackId, effectId);
+}
+
+void EffectsActionsController::removeRealtimeEffect(const muse::actions::ActionData& args)
+{
+    IF_ASSERT_FAILED(args.count() == 2) {
+        return;
+    }
+
+    const auto project = globalContext()->currentProject();
+    IF_ASSERT_FAILED(project) {
+        // Command issued without an open project ?..
+        return;
+    }
+
+    const auto trackId = args.arg<au::audio::TrackId>(0);
+    const auto effectState = args.arg<au::audio::RtEffectState>(1);
+    audioEngine()->removeRealtimeEffect(*project, trackId, effectState);
+}
+
+void EffectsActionsController::replaceRealtimeEffect(const muse::actions::ActionData& args)
+{
+    IF_ASSERT_FAILED(args.count() == 3) {
+        return;
+    }
+
+    const auto project = globalContext()->currentProject();
+    IF_ASSERT_FAILED(project) {
+        // Command issued without an open project ?..
```

Comment:
these comments are useless, you set for them `UiCtxProjectOpened` context, so this won't happen and the likelihood that someone will encounter this is low (I am not sure that someone will change their context)